packages/wallet/src/cip30.ts

```diff
@@ -122,10 +148,22 @@ export const createWalletApi = (
   },
   signData: async (addr: Cardano.Address, payload: Bytes): Promise<Cip30DataSignature> => {
     logger.debug('signData');
-    return cip30signData({
-      keyAgent: wallet.keyAgent,
-      payload: Cardano.util.HexBlob(payload),
-      signWith: addr
+    return confirmationCallback({
+      data: {
+        addr,
+        payload: Cardano.util.HexBlob(payload)
+      },
+      type: Cip30ConfirmationCallbackType.SignData
+    }).then(async (proceed: boolean) => {
```

Comment:
Since this is already an async function, let's convert this (and signTx confirmation) to `await` syntax